ui/app/AppLayouts/Chat/components/ChannelIcon.qml

```diff
@@ -0,0 +1,79 @@
+import QtQuick 2.3
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.3
+import Qt.labs.platform 1.1
+import "../../../../imports"
+import "../../../../shared"
+
+Rectangle {
+  property var topMargin: Theme.smallPadding
+  property var bottomMargin: Theme.smallPadding
+  property var channelName
+  property var channelType
+  id: contactImage
+  width: width ? width : 36
+  height: height ? height: 36
+  anchors.left: parent.left
+  anchors.leftMargin: Theme.padding
+  anchors.top: parent.top
+  anchors.topMargin: topMargin
+  anchors.bottom: parent.bottom
+  anchors.bottomMargin: bottomMargin
+  radius: 50
+
+  Loader {
+    sourceComponent: channelType == Constants.chatTypeOneToOne ? imageIdenticon : letterIdenticon
+    anchors.fill: parent
+  }
+
+  Component {
+    id: letterIdenticon
+    Rectangle {
+      width: width ? width : 36
+      height: height ? height: 36
```

Comment:
Don't think this will work. Well, it will always go back to the 36, because no `width` will ever be passed to this rectangle. 
I think what you wan to do is check if a width was passed to the ChannelIcon? If so, you want to do something like: `contactImage.width ? ...`

Reply:
Ach I can try that! I tried redefining the `height` and `width` properties which wasn't allowed. Will try the reference version you're suggesting!